software/displayerang.livecodescript

```diff
@@ -0,0 +1,1402 @@
+   -- node display label ----------------
+   repeat for each key tRec in pData["nodes"]
+      put pShort[pDoc][tRec]["title"] into tForShow
+      if tForShow is empty then
+         --         put symbolLookup(pShort[pDoc][tRec]["type"]) && tRec into tForShow
+         put "<p>" & symbolLookupHTML(pShort[pDoc][tRec]["type"]) && tRec & "</p>" into tForShow
+      else
+         --         put symbolLookup(pShort[pDoc][tRec]["type"]) & " " before tForShow
+         put symbolLookupHTML(pShort[pDoc][tRec]["type"]) into tSymbol
+         put "<p>" & tSymbol & space before tForShow
+         put "</p>" after tForShow
+      end if
+      put tForShow into pData["nodes"][tRec]["displayString"]
+      put tForShow into pData["nodes"][tRec]["displayStringExtended"]
+      -- pass forward the memo & figure if they're present
+      put pShort[pDoc][tRec]["memo"] into tForShow
+      if tForShow is not empty then put cr & "<p>" & tForShow & "</p>" after pData["nodes"][tRec]["displayStringExtended"]
+      put pShort[pDoc][tRec]["figure"] into tForShow
+      -- set all of the <img> tags to the same image object
+      -- later, if this figure needs to be shown in a field (like tooltip), that image object's filename will be set to the figure's path
+      -- the field's image won't care what's in the image object
+      -- 17624 is the ID of image "figureSource"
+      if tForShow is not empty then 
+         put cr & "<p><img src=17624></p>" after pData["nodes"][tRec]["displayStringExtended"]
+         
+      end if
+   end repeat
+end tempConvertData
+
+on middleOut pData,@pDraw,@pFocus -- ! a hacky way to draw the provided graph, calls assignLayers
+   -- ! as of 10JAN2016 middleOut only follows flow links, so it won't visit or follow dive/rise links
+   put "on middleOut" into xX["write"];    devLog xX
+   -- pData ! just sGraphData at the moment, maybe this should be an adjacency list ?
+   -- setup variables --------------------------------------------------------------------------------
+   put empty into pData["layers"]
+   put empty into pData["visited"]
+   -- make sure the focus node is actually focusable
+   put the keys of pData["nodes"] into tKeys
+   put empty into tRecs
+   repeat for each line tLine in tKeys
+      if tLine is a number then put tLine into line (the number of lines of tRecs)+1 of tRecs
+   end repeat
+   if pFocus is not among the lines of tRecs then
+      sort tRecs numeric descending
+      put line 1 of tRecs into pFocus
+   end if
+   -- recursively assign layers ------------------------------------------------------------------------
+   -- pass pData by reference, and focus node, and initial layer
+   -- add layer and visited info to pData
+   assignLayers pData,pFocus,0
+   -- remove duplicates in layers ------------------------------------------------------------------
+   repeat for each key tLayer in pData["layers"]
+      put pData["layers"][tLayer]into tOneLayer
```

Comment:
@matthewmaier Add a space after `tLayer]` incase a future complier doesn't like it